File: 2018/day01/python/solution.py
def _parse_input(input_data):
    if not isinstance(input_data, list):
        input_data = input_data.split('\n')
    return [int(v.strip()) for v in input_data]

def part1(input_data):
    moves = _parse_input(input_data)
    return sum(moves)


def part2(input_data):
    moves = _parse_input(input_data)
    total = 0
    seen = {0}

    while True:
        for move in moves:
            total += move
            if total in seen:
                return total
            seen.add(total)

File: 2018/day01/typescript/solution.ts
function _prepareMoves(input_data: string | number[]): number[] {
  if (Array.isArray(input_data)) {
    return input_data.map((v) => parseInt(v.toString().trim()));
  }

  return input_data
    .trim()
    .split("\n")
    .map((v) => parseInt(v.trim()));
}

export function part1(input_data: string): number {
  const moves = _prepareMoves(input_data);
  const total = moves.reduce((sum, move) => sum + move, 0);

  return total;
}

export function part2(input_data: string): number {
  const moves = _prepareMoves(input_data);

  let total = 0;
  const seen = new Set<number>();
  seen.add(0);

  while (true) {
    for (const move of moves) {
      total += move;

      if (seen.has(total)) {
        return total;
      }

      seen.add(total);
    }
  }
}
